tests/integration/ie-tests.sh

```diff
@@ -1,6 +1,17 @@
 #!/bin/bash
 
-readonly SPARK_IMAGE='ghcr.io/canonical/charmed-spark:3.4-22.04_edge'
+# Copyright 2024 Canonical Ltd.
+
+# Import reusable utilities
+source ./tests/integration/utils/s3-utils.sh
+source ./tests/integration/utils/azure-utils.sh
+
+
+readonly SPARK_IMAGE='ghcr.io/canonical/charmed-spark@sha256:fd458fbbe4b90232ec657c92375da01789396f914865bcb4c0f919bf33c031f4'
```

Comment:
**todo** now that the debugging is done, we can revert to NOT pin the image, and use `3.4-22.04_edge`